Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/HDRColorGrading.azsl

```diff
@@ -0,0 +1,212 @@
+// it will be converted into ACEScg as needed.
+float3 ColorGradePostExposure (float3 frame_color, float exposure) 
+{
+    frame_color *= pow(2.0f, exposure);
+	return frame_color;
+}
+
+// The contrast equation is performed in ACEScc (logarithmic) color space.
+float3 ColorGradingContrast (float3 frame_color, float midgrey, float amount) 
+{
+    float contrast_adjustment = amount * 0.01f + 1.0f;
+    frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScg, ColorSpaceId::ACEScc);
+    frame_color = (frame_color - midgrey) * contrast_adjustment + midgrey;
+	return frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScc, ColorSpaceId::ACEScg);
+}
+
+// The swatch_color param expects a linear RGB value.
+float3 ColorGradeColorFilter (float3 frame_color, float3 swatch_color, float alpha) 
+{
+    swatch_color = TransformColor(frame_color.rgb, ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    swatch_color *= pow(2.0f, PassSrg::m_colorFilterIntensity);
+    float3 frame_adjust = frame_color * swatch_color;
+    return frame_color = lerp(frame_color, frame_adjust, alpha);
+}
+
+float3 ColorGradeHueShift (float3 frame_color, float amount) 
+{
+	float3 frame_hsv = TransformColor(frame_color, ColorSpaceId::ACEScg, ColorSpaceId::HSV);
+	float hue = frame_hsv.x + amount;
+	frame_hsv.x = RotateHue(hue, 0.0, 1.0);
+	return TransformColor(frame_hsv, ColorSpaceId::HSV, ColorSpaceId::ACEScg);
+}
+
+float3 ColorGradeSaturation (float3 frame_color, float control) 
+{
+	float v_luminance = CalculateLuminance(frame_color, ColorSpaceId::ACEScg);
+	return (frame_color - v_luminance) * control + v_luminance;
+}
+
+float3 ColorGradeKelvinColorTemp(float3 frame_color, float kTemp, float mix, float lumPreservation)
+{
+    float3 kColor    = TransformColor(KelvinToRGB(kTemp), ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    float3 oLum      = RGBToHSL(frame_color.rgb);
+    float3 blended   = lerp(frame_color.rgb, frame_color.rgb * kColor.rgb, mix);
+    float3 resHSV    = RGBToHSL(blended.rgb);
+    float3 resRGB    = HSLToRGB(float3(resHSV.xy, oLum.z));
+    return lerp(blended.rgb, resRGB.rgb, lumPreservation);
+}
+
+// pow(f, e) won't work if f is negative, or may cause inf/NAN.
+float3 NoNanPow(float3 base, float3 power)
+{
+    return pow(max(abs(base), float3(FLOAT_EPSILON, FLOAT_EPSILON, FLOAT_EPSILON)), power);
+}
+
+float3 ColorGradeSplitTone (float3 frame_color, float balance, float mix) 
+{
+    float3 frame_splitTone = NoNanPow(frame_color, 1.0 / 2.2);
+	float t = saturate(CalculateLuminance(saturate(frame_splitTone), ColorSpaceId::ACEScg) + balance);
+	float3 shadows = lerp(0.5, PassSrg::m_splitToneShadowsColor.rgb, 1.0 - t);
+	float3 highlights = lerp(0.5, PassSrg::m_splitToneHighlightsColor.rgb, t);
+	frame_splitTone = BlendMode_SoftLight(frame_splitTone, shadows);
+	frame_splitTone = BlendMode_SoftLight(frame_splitTone, highlights);
+    frame_splitTone = NoNanPow(frame_splitTone, 2.2);
```

Comment:
nit: spacing

Reply:
hmm tabs were used for some of the files in this PR instead of spaces. Let me fix that.